File: leetcode/0661. Image Smoother.java
class Solution {
    public int[][] imageSmoother(int[][] img) {
        int[][] outputImg = new int[img.length][img[0].length];
        for(int r = 0; r < img.length; ++r){
            for(int c = 0; c < img[0].length; ++c){
                outputImg[r][c] = getFilteredValue(img, r, c);
            }
        }
        return outputImg;
    }

    private int getFilteredValue(int[][] img, int x, int y){
        int m = img.length, n = img[0].length;
        double sum = 0, count = 0;
        for(int r = Math.max(0, x - 1); r < Math.min(m , x + 2); r++){
            for(int c = Math.max(0, y - 1); c < Math.min(n, y + 2); c++){
                sum += img[r][c];
                ++count;
            }
        }
        return (int) Math.floor(sum / count);
    }
}

File: leetcode/0661. Image Smoother.py
class Solution:
    def imageSmoother(self, img: List[List[int]]) -> List[List[int]]:
        return [
            [self.getFiteredValue(img, i, j) for j in range(len(img[0]))] 
            for i in range(len(img))
        ]
        
    def getFiteredValue(self, img: List[List[int]], x: int, y: int) -> int:
        tmp = [
            img[r][c]
            for c in range(max(0, y - 1), min(len(img[0]), y + 2)) 
            for r in range(max(0, x - 1), min(len(img), x + 2))
        ]
        return sum(tmp) // len(tmp)